app/models/category.rb

```diff
@@ -795,8 +794,12 @@ def self.find_by_slug_path(slug_path)
     return nil if slug_path.empty?
     return nil if slug_path.size > SiteSetting.max_category_nesting
 
-    if SiteSetting.slug_generation_method == "encoded"
-      slug_path.map! { |slug| CGI.escape(slug) }
+    slug_path.map! do |slug|
+      if SiteSetting.slug_generation_method == "encoded"
+        CGI.escape(slug.downcase)
```

Comment:
Why not do the `downcase` after the `CGI.escape`? Does it have a different result?

Reply:
If I remember correctly, doing the `downcase` operation after `CGI.escape` did not work for all strings.